apps/prairielearn/python/conftest.py

```diff
@@ -88,7 +88,7 @@ def r_types_dataframe() -> pd.DataFrame:
             "logical-list": pd.Series([True, False, True]),
             "character-string-list": pd.Series(["a", "b", "c"]),
             # Time Dependency: https://pandas.pydata.org/docs/user_guide/timeseries.html
-            "POSIXct-POSIXt-timestamp": pd.Timestamp("20230102"),
+            "POSIXct-POSIXt-timestamp": pd.Timestamp("2023-01-02T00:00:00.0000000"),
```

Comment:
Can you point to a Pandas changelog item explaining why this had to change? Is there any way that this could negatively impact instructor code?

Reply:
In short, the constructor for the `Timestamp` object was changed: https://pandas.pydata.org/docs/whatsnew/v2.1.0.html#datetimelike

Basically, the type of the datetime object that gets created here has a precision as part of its type (seconds, milliseconds, nanoseconds, etc.). Before, the constructor was always turning this into a `datetime[ns]`, but the new behavior after the version bump was turning this into a `datetime[s]`. Then, during the test, this caused a type mismatch because, after reserializing the dataframe with `pl.from_json`, timestamps always get turned into a `datetime[ns]`, even though the values are actually the same. So adding the extra zeros here causes the type to be `datetime[ns]`.

I think it's very unlikely that any instructor is going to have issues with this (as this was actually an increase in precision), and it basically only came up because the test cases check the exact data type in addition to comparing the values.